[WFLY-4692] : Configure the distributions in the dist modules instead of the feature packs

ehsavoie opened the pull request:
Using resources plugin to add the product.conf and product module when building the distributions instead of packing them with the feature packs.

Jira: https://issues.jboss.org/browse/WFLY-4692

jamezp commented on web-dist/pom.xml:
```diff
@@ -56,7 +56,9 @@
 
-
+    <properties>
+        <product.release.name>WildFly Servlet</product.release.name>
```
We should get @n1hility to okay the name, but it looks good to me.

ehsavoie committed:
[WFLY-4692] : Configure the distributions in the dist modules instead of the feature packs.

Using resources plugin to add the product.conf and product module when building the distributions instead of packing them with the feature packs.
